wasm/sdk.go

```diff
@@ -0,0 +1,1178 @@
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetQuerySleepTime(num)
+	return nil
+}
+
+func SetMinSubmit(this js.Value, p []js.Value) interface{} {
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetMinSubmit(num)
+	return nil
+}
+
+func SetMinConfirmation(this js.Value, p []js.Value) interface{} {
+	num, _ := strconv.Atoi(p[0].String())
+	sdk.SetMinConfirmation(num)
+	return nil
+}
+
+func ZBOXSetNetwork(this js.Value, p []js.Value) interface{} {
+	var miners []string
+	var sharders []string
+	jsMiners := p[0]
+	jsSharders := p[1]
+
+	if got := js.Global().Get("Array").Call("isArray", jsMiners).Bool(); got {
+		for i := 0; i < jsMiners.Length(); i++ {
+			if got := jsMiners.Index(i).Type().String(); got == "string" {
+				miners = append(miners, jsMiners.Index(i).String())
+			} else {
+				return map[string]interface{}{
+					"error": fmt.Sprintf("SetNetwork failed. Reason: expected type \"string\". got=%#v", jsMiners.Index(i).Type().String()),
+				}
+			}
+		}
+	}
+
+	if got := jsMiners.Type().String(); got == "string" {
+		miners = append(miners, jsMiners.String())
+	}
+
+	if got := js.Global().Get("Array").Call("isArray", jsSharders).Bool(); got {
+		for i := 0; i < jsSharders.Length(); i++ {
+			if got := jsSharders.Index(i).Type().String(); got == "string" {
+				sharders = append(sharders, jsSharders.Index(i).String())
+			} else {
+				return map[string]interface{}{
+					"error": fmt.Sprintf("SetNetwork failed. Reason: expected type \"string\". got=%#v", jsSharders.Index(i).Type().String()),
+				}
+			}
+		}
+	}
+
+	if got := jsSharders.Type().String(); got == "string" {
+		sharders = append(sharders, jsSharders.String())
+	}
+
+	if len(miners) > 0 && len(sharders) > 0 {
+		sdk.SetNetwork(miners, sharders)
+	}
+
+	return map[string]interface{}{
+		"error": "SetNetwork failed. Reason: empty miners or sharders",
+	}
+}
+
+// //
+// // read pool
+// //
+
+func CreateReadPool(this js.Value, p []js.Value) interface{} {
+	handler := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+		resolve := args[0]
+		reject := args[1]
+
+		go func() {
+			err := sdk.CreateReadPool()
+			if err != nil {
+				reject.Invoke(map[string]interface{}{
+					"error": fmt.Sprintf("CreateReadPool failed. Reason: %s", err),
+				})
+			}
+			resolve.Invoke(map[string]interface{}{
+				"result": fmt.Sprintf("%s", err),
+			})
+		}()
+
+		return nil
+	})
+
+	promiseConstructor := js.Global().Get("Promise")
+	return promiseConstructor.New(handler)
+}
+
+func AllocFilter(this js.Value, p []js.Value) interface{} {
+	poolStats := p[0].String()
+	allocID := p[1].String()
+
+	var alloc sdk.AllocationPoolStats
+	err := json.Unmarshal([]byte(poolStats), &alloc)
+	if err != nil {
+		return map[string]interface{}{
+			"error": fmt.Sprintf("AllocFilter failed. Reason: %s", err),
+		}
+	}
+	allocFilter := (*sdk.AllocationPoolStats).AllocFilter
+
+	allocFilter(&alloc, allocID)
+	return nil
+}
+
+func ZBOXGetReadPoolInfo(this js.Value, p []js.Value) interface{} {
```

Comment:
Please, refactor names without ZBOX

Reply:
@moldis It has name collition with the `wasm/wallet.go` function. What do you suggest? Because the function should be the same as in the `zboxcore/sdk.go` function which is `GetReadPoolInfo` .

Reply:
So I removed all `ZBOX` word in `wasm/sdk.go` function and changed all function collision in `wasm/wallet.go` by adding `Wallet` word after the Get/Set .